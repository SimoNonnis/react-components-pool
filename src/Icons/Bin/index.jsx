import React from 'react';
import PropTypes from 'prop-types';

function Bin({ width, height, fill, className, classNamePath }) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 512 512"
      width={width}
      height={height}
      className={className}
    >
      <title>Bin Icon</title>

      <path
        className={classNamePath}
        fill={fill}
        d="M436,60H346V45A45.05,45.05,0,0,0,301,0H211a45.05,45.05,0,0,0-45,45V60H76a45.05,45.05,0,0,0-45,45v30a15,15,0,0,0,15,15H62.18L88.57,471s0,.08,0,.12A44.78,44.78,0,0,0,133.4,512H378.6a44.78,44.78,0,0,0,44.82-40.93s0-.08,0-.12L449.82,150H466a15,15,0,0,0,15-15V105A45.05,45.05,0,0,0,436,60ZM196,45a15,15,0,0,1,15-15h90a15,15,0,0,1,15,15V60H196ZM393.54,468.41A14.93,14.93,0,0,1,378.6,482H133.4a14.93,14.93,0,0,1-14.94-13.59L92.28,150H419.72ZM451,120H61V105A15,15,0,0,1,76,90H436a15,15,0,0,1,15,15Z"
      />

      <path
        className={classNamePath}
        fill={fill}
        d="M256,180a15,15,0,0,0-15,15V407a15,15,0,0,0,30,0V195A15,15,0,0,0,256,180Z"
      />
      <path
        className={classNamePath}
        fill={fill}
        d="M346,180a15,15,0,0,0-15,15V407a15,15,0,0,0,30,0V195A15,15,0,0,0,346,180Z"
      />
      <path
        className={classNamePath}
        fill={fill}
        d="M166,180a15,15,0,0,0-15,15V407a15,15,0,0,0,30,0V195A15,15,0,0,0,166,180Z"
      />
    </svg>
  );
}

Bin.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  fill: PropTypes.string,
  className: PropTypes.string,
  classNamePath: PropTypes.string,
};

Bin.defaultProps = {
  width: 40,
  height: 40,
  fill: 'currentColor',
  className: 'sizeIcon',
  classNamePath: 'sizeIconPath',
};

export default Bin;
