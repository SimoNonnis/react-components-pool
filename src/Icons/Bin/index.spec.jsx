/* eslint-env jest */
import React from 'react';

import Bin from './';

describe('Test <Bin /> icon', () => {
  it('should render Bin icon correctly with default props', () => {
    const wrapper = shallow(<Bin />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Bin icon correctly with props', () => {
    const wrapper = shallow(
      <Bin fill="#a4a1a1" className="testClassname" height={30} width={30} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
