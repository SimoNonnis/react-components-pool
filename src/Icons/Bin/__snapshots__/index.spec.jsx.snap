// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Bin /> icon should render Bin icon correctly with default props 1`] = `
<svg
  className="sizeIcon"
  height={40}
  viewBox="0 0 512 512"
  width={40}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Bin Icon
  </title>
  <path
    className="sizeIconPath"
    d="M436,60H346V45A45.05,45.05,0,0,0,301,0H211a45.05,45.05,0,0,0-45,45V60H76a45.05,45.05,0,0,0-45,45v30a15,15,0,0,0,15,15H62.18L88.57,471s0,.08,0,.12A44.78,44.78,0,0,0,133.4,512H378.6a44.78,44.78,0,0,0,44.82-40.93s0-.08,0-.12L449.82,150H466a15,15,0,0,0,15-15V105A45.05,45.05,0,0,0,436,60ZM196,45a15,15,0,0,1,15-15h90a15,15,0,0,1,15,15V60H196ZM393.54,468.41A14.93,14.93,0,0,1,378.6,482H133.4a14.93,14.93,0,0,1-14.94-13.59L92.28,150H419.72ZM451,120H61V105A15,15,0,0,1,76,90H436a15,15,0,0,1,15,15Z"
    fill="currentColor"
  />
  <path
    className="sizeIconPath"
    d="M256,180a15,15,0,0,0-15,15V407a15,15,0,0,0,30,0V195A15,15,0,0,0,256,180Z"
    fill="currentColor"
  />
  <path
    className="sizeIconPath"
    d="M346,180a15,15,0,0,0-15,15V407a15,15,0,0,0,30,0V195A15,15,0,0,0,346,180Z"
    fill="currentColor"
  />
  <path
    className="sizeIconPath"
    d="M166,180a15,15,0,0,0-15,15V407a15,15,0,0,0,30,0V195A15,15,0,0,0,166,180Z"
    fill="currentColor"
  />
</svg>
`;

exports[`Test <Bin /> icon should render Bin icon correctly with props 1`] = `
<svg
  className="testClassname"
  height={30}
  viewBox="0 0 512 512"
  width={30}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Bin Icon
  </title>
  <path
    className="sizeIconPath"
    d="M436,60H346V45A45.05,45.05,0,0,0,301,0H211a45.05,45.05,0,0,0-45,45V60H76a45.05,45.05,0,0,0-45,45v30a15,15,0,0,0,15,15H62.18L88.57,471s0,.08,0,.12A44.78,44.78,0,0,0,133.4,512H378.6a44.78,44.78,0,0,0,44.82-40.93s0-.08,0-.12L449.82,150H466a15,15,0,0,0,15-15V105A45.05,45.05,0,0,0,436,60ZM196,45a15,15,0,0,1,15-15h90a15,15,0,0,1,15,15V60H196ZM393.54,468.41A14.93,14.93,0,0,1,378.6,482H133.4a14.93,14.93,0,0,1-14.94-13.59L92.28,150H419.72ZM451,120H61V105A15,15,0,0,1,76,90H436a15,15,0,0,1,15,15Z"
    fill="#a4a1a1"
  />
  <path
    className="sizeIconPath"
    d="M256,180a15,15,0,0,0-15,15V407a15,15,0,0,0,30,0V195A15,15,0,0,0,256,180Z"
    fill="#a4a1a1"
  />
  <path
    className="sizeIconPath"
    d="M346,180a15,15,0,0,0-15,15V407a15,15,0,0,0,30,0V195A15,15,0,0,0,346,180Z"
    fill="#a4a1a1"
  />
  <path
    className="sizeIconPath"
    d="M166,180a15,15,0,0,0-15,15V407a15,15,0,0,0,30,0V195A15,15,0,0,0,166,180Z"
    fill="#a4a1a1"
  />
</svg>
`;
